src/apps/external_executor/external_executor.cpp

```diff
@@ -689,14 +728,29 @@ namespace externalexecutor
       return std::make_shared<ExternallyExecutedEndpoint>();
     }
 
+    ExecutorId validate_supported_endpoints(
+      ccf::endpoints::EndpointContext& endpoint_ctx)
+    {
+      std::string method = endpoint_ctx.rpc_ctx->get_request_verb().c_str();
+      std::string uri = endpoint_ctx.rpc_ctx->get_request_path();
+
+      if (supported_uris.find(method + uri) == supported_uris.end())
+      {
+        throw std::logic_error("Only registered endpoints are supported");
+      }
+      auto executor_id = supported_uris[method + uri].get_executor_id();
```

Comment:
We should store the iterator here to avoid doing the lookup twice.
```suggestion
      auto it = supported_uris.find(method + uri);
      if (it == supported_uris.end())
      {
        throw std::logic_error("Only registered endpoints are supported");
      }
      auto executor_id = it->second.get_executor_id();
```

May be `it.second->get` at the end there, I forget which way round that goes.

Also, should report the unsupported path in the error, something like "Could not find any executors supporting {method} {uri}".